src/main/scala/com/microsoft/partnercatalyst/fortis/spark/Pipeline.scala

```diff
@@ -0,0 +1,100 @@
+package com.microsoft.partnercatalyst.fortis.spark
+
+// TODO: remove this once transform context settings can be read from Cassandra
+import com.microsoft.partnercatalyst.fortis.spark.ProjectFortis.Settings
+
+import scala.reflect.runtime.universe.TypeTag
+import com.microsoft.partnercatalyst.fortis.spark.analyzer.{Analyzer, ExtendedFortisEvent}
+import com.microsoft.partnercatalyst.fortis.spark.dba.ConfigurationManager
+import com.microsoft.partnercatalyst.fortis.spark.dto.{Analysis, FortisEvent}
+import com.microsoft.partnercatalyst.fortis.spark.streamprovider.StreamProvider
+import com.microsoft.partnercatalyst.fortis.spark.transforms.ZipModelsProvider
+import com.microsoft.partnercatalyst.fortis.spark.transforms.image.{ImageAnalysisAuth, ImageAnalyzer}
+import com.microsoft.partnercatalyst.fortis.spark.transforms.language.{LanguageDetector, LanguageDetectorAuth}
+import com.microsoft.partnercatalyst.fortis.spark.transforms.locations.{Geofence, LocationsExtractor, LocationsExtractorFactory, PlaceRecognizer}
+import com.microsoft.partnercatalyst.fortis.spark.transforms.locations.client.FeatureServiceClient
+import com.microsoft.partnercatalyst.fortis.spark.transforms.people.PeopleRecognizer
+import com.microsoft.partnercatalyst.fortis.spark.transforms.sentiment.{SentimentDetector, SentimentDetectorAuth}
+import com.microsoft.partnercatalyst.fortis.spark.transforms.topic.KeywordExtractor
+import org.apache.spark.streaming.StreamingContext
+import org.apache.spark.streaming.dstream.DStream
+
+object Pipeline {
+  def apply[T: TypeTag](name: String, analyzer: Analyzer[T], ssc: StreamingContext, streamProvider: StreamProvider, configurationManager: ConfigurationManager): Option[DStream[FortisEvent]] = {
+    val configs = configurationManager.fetchStreamConfiguration(name)
+    val sourceStream = streamProvider.buildStream[T](ssc, configs)
+
+    sourceStream.map(_.transform(rdd => {
+
+      // TODO: populate these local variables with values from global update/broadcast wrappers,
+      // which will use the ConfigurationManager (cassandra) to update themselves and (where applicable)
+      // broadcast changes across the Spark cluster.
+
+      val geofence = Geofence(north = 49.6185146245, west = -124.9578052195, south = 46.8691952854, east = -121.0945042053)
```

Comment:
That's a great question, with a non-obvious answer that has to do with the serialization nuances of checkpointing. The reason for doing this is because we need to poll Cassandra here and possibly update the transform tooling (with new settings etc.). This has to be a polling-based approach from within the DStream definition itself since when Spark comes back up from a checkpoint, we will no longer be able to affect any of its closures. The closure of the `transform` function here is serialized to the checkpoint, so when Spark comes back up, the checkpoint must know how to update itself.

Instead of `new`ing up instances of the transform tools here directly, we will call into wrapper objects which will recycle or regenerate (and rebroadcast where applicable) the instances of these tools depending on if there is a configuration change. This will happen once per batch on the driver program (so maybe once every 5 to 10 seconds).